conn.go

```diff
@@ -147,10 +147,10 @@ func (c *Conn) Close() error {
 
 	c.closed = true
 
-	// close streams
+	// reset streams
 	streams := c.Streams()
 	for _, s := range streams {
-		s.Close()
+		s.Reset()
```

Comment:
Shouldn't that be the responsibility of the underlying stream-muxer? A stream-muxer probably has a map of streams (e.g. to enforce limits of the maximum number of concurrent streams). QUIC does that already, not sure about SPDY and Yamux though.